net/ieee802154/ieee802154_sockif.c

```diff
@@ -381,9 +381,11 @@ static int ieee802154_bind(FAR struct socket *psock,
       return -EBADF;
     }
 
+  conn = (FAR struct ieee802154_conn_s *)psock->s_conn;
```

Comment:
There are still many places where psock->s_conn is introduced w/o DEBUGASSERT(conn != NULL). Do you think the debug asserts are not reasonable for them?